### DISCUSSION TITLE
SML show MeterID as text instead of hex string

### DISCUSSION THREAD

**Discussion by FerFemNemBem** [arendst/Tasmota#16232] [Q&A] [answered]
Hi,

if I want to have the meter-maufacturer shown inside the web interface it would make sense to convert the hex string to ASCII.
Example:
`1,77078181c78203ff@#,Service ID,,Meter_id,0`
ends in
`"45425a"` in the tasmota web interface.
As `"45425a"` represents "EBZ" in ASCII (the manufacturer name) it would be nice to have the possibility to convert it somehow and show the result inside the web interface instead of the raw hex as string.

Maybe something where @gemu2015 could help?

Best regards,
FFNB

**Comment by gemu2015** [comment, marked as answer]
added this option, just append an 'x' to the '#' char :
1,77078181c78203ff@#x,Service ID,,Meter_id,0

**Reply by FerFemNemBem** [reply]
Hi @gemu2015 
works like a charm. There is a dot at the end that may not be there but that doesn't matter (for me at least).
![grafik](https://user-images.githubusercontent.com/74377239/185114797-c6c012ea-d621-41c3-be41-d53718686f5e.png)

Sript:
`1,77078181c78203ff@#x,Service ID,,Meter_id,0`
Debug:
`77078181c78203ff010101010445425a01`

Thank you!

Best regards,
FFNB

**Reply by gemu2015** [reply]
ok try this one:
[xsns_53_sml.ino.zip](https://github.com/arendst/Tasmota/files/9363378/xsns_53_sml.ino.zip)

**Reply by FerFemNemBem** [reply]
Hi @gemu2015 
looks great: 
![grafik](https://user-images.githubusercontent.com/74377239/185175310-c92546f5-041b-4c11-9378-3e9150041f9d.png)

Thanks again.

Best regards,
FFNB

**Reply by Oktonaut** [reply]
I tried the modified xsns_53_sml.ino... but only works for me with the manufacturer ID `7707010060320101` but unfortunately not with the device ID `77070100600100ff`

**Reply by gemu2015** [reply]
this option is only meant for HEX coded ASCI chars as in example above
post your device ID in hex.
and there is only one string allowed per meter, so you cannot show both.

**Reply by Oktonaut** [reply]
I use the current Tasmota 12.1.1

The declaration with `#x` only works with the modified xsns_53_sml.ino and only with the manufacturer ID.
Since only one string is supported per counter, I naturally tried querying the different values one after the other.

My meter is a [DWS7412.2.G2](https://docplayer.org/132337135-Feb-dvs74-handbuch-elektronischer-elektrizitaetszaehler-dxs74-und-dxs74-typ-g2-dzg-metering-gmbh-heidelberger-str-32-d-oranienburg.html) from DZG (Deutsche Zahlergesellschaft) and these three letters from the manufacturer are also converted correctly: `445a47` = _**DZG**_
From the meter ID: `0a01445a4700026443e1` only this component is output, the rest not.
It is obviously a combination of decimal and ASCII values, because the correct number would probably be: _2561**DZG**040125409_
Although I don't know what _2561_ should stand for!? Normally, according to [DIN 43863-5](https://netze.estw.de/erlangenGips/Erlangen/__attic__20210120_155237__estw1.de/Kopfnavigation/Netze/Messwesen/Messwesen/Herstelleruebergreifende-Identifikationsnummer-fuer-Messeinrichtungen.pdf), the identifier of the division should be at this point... So here is actually the _**1**_ for electricity (Would fit if `0a` stands for a LF)

P.S. [Manufacturer ID List](https://www.dlms.com/flag-id/flag-id-list)

**Comment by FerFemNemBem** [comment]
Hi @gemu2015 ,
are there any plans to pull this change into the standard tasmota sensor driver?
Best regards,
FFNB

**Reply by gemu2015** [reply]
ok just made a pr.

**Reply by FerFemNemBem** [reply]
Great! Thanks again!

Best regards,
FFNB

**Comment by McR1983** [comment]
Hey is it in the new Firmware included?
What is to do with the .ino ?

Thanks

**Reply by gemu2015** [reply]
it is now included in tasmota

**Reply by McR1983** [reply]
ok cool
Coud you tell me wich file i have to use for sml Ir Zähler for OTA

**Comment by gemu2015** [comment]
you have to compile your own binary

**Reply by McR1983** [reply]
Hi @gemu2015 , sorry only the "EBZ" comes out not the Number after...

In the script is this: 1,77070100000009FF@#x,Identifikation,,96_1_0,0

Without #x --> "090145425a010012xxxx"

In the first post said "As "45425a" represents "EBZ" " yes but the Number after not here

On the ESP:arendst/Tasmota/tree/v12.4

**Comment by McR1983** [comment]
Hi @gemu2015 can you make a request please
